JavaScript/JavaScript.sublime-syntax

```diff
@@ -2066,52 +2057,22 @@ contexts:
       scope: invalid.illegal.numeric.decimal.js
       pop: 1
 
-  literal-call:
-    - match: (?={{identifier_name}}\s*(?:{{dot_accessor}})?\()
-      set:
-        - call-function-meta
-        - function-call-arguments
-        - literal-variable
-
-    - match: (?={{identifier_name}}\s*(?:{{dot_accessor}}\s*#?{{identifier_name}}\s*)+(?:{{dot_accessor}})?\()
-      set:
-        - call-method-meta
-        - function-call-arguments
-        - call-path
-        - literal-variable
-
-  call-path:
-    - match: '{{dot_accessor}}'
-      scope: punctuation.accessor.js
-      push: object-property
-    - include: else-pop
-
-  call-function-meta:
-    - meta_include_prototype: false
-    - meta_scope: meta.function-call.js
-    - include: immediately-pop
-
-  call-method-meta:
-    - meta_include_prototype: false
-    - meta_scope: meta.function-call.method.js
-    - include: immediately-pop
-
   call-function-name:
     - match: '{{dollar_only_identifier}}'
-      scope: variable.function.js variable.other.dollar.only.js punctuation.dollar.js
+      scope: meta.function-call.js variable.function.js variable.other.dollar.only.js punctuation.dollar.js
       pop: 1
     - match: '{{identifier_name}}'
-      scope: variable.function.js
+      scope: meta.function-call.js variable.function.js
       pop: 1
     - include: else-pop
 
   call-method-name:
     - include: support-property
     - match: '{{identifier_name}}'
-      scope: variable.function.js
+      scope: meta.function-call.js variable.function.js
       pop: 1
     - match: '(#){{identifier_name}}'
-      scope: variable.function.js
+      scope: meta.function-call.js variable.function.js
```

Comment:
Would it make sense to merge the two patterns

```yml
    - match: '{{identifier_name}}'
      scope: meta.function-call.js variable.function.js
      pop: 1
    - match: '(#){{identifier_name}}'
      scope: meta.function-call.js variable.function.js
      pop: 1
```

to

```yml
    - match: '(#)?{{identifier_name}}'
      scope: meta.function-call.js variable.function.js
      captures:
        1: punctuation.definition.js
      pop: 1
```

?

Reply:
Agreed, done.